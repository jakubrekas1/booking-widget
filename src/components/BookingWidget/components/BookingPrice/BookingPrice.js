import React from 'react';
import PropTypes from 'prop-types';

const BookingPrice = ({ amount, currency }) => (
  <div className="booking-price">
    <p className="booking-price__value">
      {`${amount} ${currency} `}
      <span className="booking-price__label">per night</span>
    </p>
  </div>
);

BookingPrice.defaultProps = {
  currency: 'z≈Ç',
};

BookingPrice.propTypes = {
  amount: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
  currency: PropTypes.string,
};

export default BookingPrice;
