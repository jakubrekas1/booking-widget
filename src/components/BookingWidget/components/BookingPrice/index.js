export { default as BookingPrice } from './BookingPrice';
