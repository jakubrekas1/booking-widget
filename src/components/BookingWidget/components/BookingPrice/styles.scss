@import '../../../../assets/styles/colors';

.booking-price{
  font-size: 28px;
  font-weight: bold;
  color: $black;
}
