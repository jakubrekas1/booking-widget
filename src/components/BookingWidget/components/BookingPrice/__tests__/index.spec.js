import React from 'react';
import renderer from 'react-test-renderer';
import BookingPrice from '../BookingPrice';

describe('Booking price component', () => {
  it('should renders correctly', () => {
    const Component = <BookingPrice amount={256} currency="PLN" />;
    const tree = renderer.create(Component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should renders correctly with default props', () => {
    const Component = <BookingPrice amount={256} />;
    const tree = renderer.create(Component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
