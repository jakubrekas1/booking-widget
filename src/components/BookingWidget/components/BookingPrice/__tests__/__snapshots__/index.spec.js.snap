// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booking price component should renders correctly 1`] = `
<div
  className="booking-price"
>
  <p
    className="booking-price__value"
  >
    256 PLN 
    <span
      className="booking-price__label"
    >
      per night
    </span>
  </p>
</div>
`;

exports[`Booking price component should renders correctly with default props 1`] = `
<div
  className="booking-price"
>
  <p
    className="booking-price__value"
  >
    256 z≈Ç 
    <span
      className="booking-price__label"
    >
      per night
    </span>
  </p>
</div>
`;
